src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java21/ExpectedStringTemplateBasic.txt

```diff
@@ -81,10 +81,8 @@ COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
         |   |       `--DOT -> . [16:26]
         |   |           |--IDENT -> STR [16:23]
         |   |           `--STRING_TEMPLATE_BEGIN -> " [16:27]
-        |   |               |--STRING_TEMPLATE_CONTENT ->  [16:28]
```

Comment:
we need to update https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#STRING_TEMPLATE_CONTENT 
only single token example.
we need one more compact example,  to show that STRING_TEMPLATE_CONTENT might not be present.

We are very inconsistent on presence and not presence.
example is `MODIFIERS` but there are more:
```
 VARIABLE_DEF -> VARIABLE_DEF
  |--MODIFIERS -> MODIFIERS
  |--TYPE -> TYPE
  |   `--IDENT -> String
  |--IDENT -> s
```

it is always tricky to deal with it.
I am not sure what is right way to make AST: 1) always define 2) skip if empty.
I am ok to skip. 
If it was not obvious for us, better to explicitly show it.

Reply:
My issue with it was that we call it "content" but it is an imaginary node that has no content, with the same line/column number that it's next sibling does. This is nonsense.

Reply:
Example is updated